test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -735,5 +735,99 @@ public void UninstallDeprecatedPackageFromUI(ProjectTemplate projectTemplate, st
+
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreClassLib, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+            solutionService.Build();
+
+            // Assert
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, TestPackageName, TestPackageVersionV1, Logger);
+            uiwindow.AssertPackageNotTransitive();
+
+            // Act (Search the transitive package since it will not show at the top of package list by default)
+            uiwindow.SearchPackageFromUI(transitivePackageName);
+
+            // Assert
+            VisualStudio.AssertNoErrors();
+            uiwindow.AssertPackageTransitive();
+        }
+
+        [TestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task InstallTransitivePackageFromUI()
+        {
+            // Arrange
+            var transitivePackageName = "Contoso.B";
+            await CommonUtility.CreateDependenciesPackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1, transitivePackageName, TestPackageVersionV1);
+
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+            uiwindow.InstallPackageFromUI(transitivePackageName, TestPackageVersionV1);
+            uiwindow.SearchPackageFromUI(transitivePackageName);
+            solutionService.Build();
+
+            // Assert
+            VisualStudio.AssertNoErrors();
+            uiwindow.AssertPackageNotTransitive();
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, transitivePackageName, TestPackageVersionV1, Logger);
+        }
+
+        [TestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task UninstallTransitivePackageFromUI()
```

Comment:
I'm not sure this does anything for us. 

1. Install package with transitive dep
2. Install that transitive dep as top level

The 2nd package being a transitive dependency at some point doesn't really affect how uninstall works, at least not in a .NET SDK based projects

Reply:
No, we are talking about different scenarios. Sorry about the confusion. In this case, firstly we installed a top-level package which having a dependency package. Then we installed the transitive package (that will change it into a top-level package). The purpose of this case is testing uninstallation of the top-level package (which was a transitive package before) will make it back to transitive package from top-level package.

Reply:
Ah, I get it now, thanks for the explanation. 

Naming wise, I'm not really sure the current test name tells the whole story. Maybe we need to workshop that a little bit.